tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/KeramelizerInputLanguagePred.scala

```diff
@@ -0,0 +1,97 @@
+package at.forsyte.apalache.tla.lir.transformations.standard
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.oper.ApalacheOper.gen
+import at.forsyte.apalache.tla.lir.oper._
+import at.forsyte.apalache.tla.lir.transformations.{PredResult, PredResultFail, PredResultOk}
+
+import scala.collection.immutable.HashSet
+
+/**
+ * Test whether expressions fit into the input fragment of [[Keramelizer]], i.e., whether an expression
+ *   - is flattened (see [[FlatLanguagePred]]),
+ *   - operators `exists`, `forall`, `chooseBounded`, `filter`, `apply` take a [[NameEx]] as first argument, and
+ *   - `Seq(_)` is prohibited.
+ *
+ * To get a better idea of the accepted fragment, check [[TestKeramelizerInputLanguagePred]].
+ */
+class KeramelizerInputLanguagePred extends ContextualLanguagePred {
```

Comment:
Something I don't quite understand: if these preconditions are all necessary for valid input for Keramelizer, then why would this check but put into the `apply` method of the `Keramelizer` itself?

Reply:
I assume that nothing violating these constraints reaches `Keramelizer` in our toolchain. @Kukovec @konnov can you confirm?
As per my offline discussion with @konnov, I assumed `KeraLanguagePred` as ground truth, i.e., that if `Keramelizer(x)` isn't recognized by `KeraLanguagePred`, then `x` should not be legal input to Keramelizer.

Reply:
So I'm still slightly unclear on the reasoning for why we'd need this extra predicate to guard inputs for the tests. Seems to me that all the logic for validation should either already be in `KreaLanguagePred`, or else we'd have some automated way of reducing expressions to valid valid Keramelizer input, and the generated expressions could be run through that?

But this may just be to lack of context on my part. So don't want to block on this, esp. since Igor has already approved :)  Please merge whenever you're satisfied on this point.

Reply:
My understanding is that these expressions wouldn't go through SANY, but that's a heavy dependency for a unit test.
I will merge this as-is to avoid lagging too far behind `unstable`.